### DISCUSSION TITLE
Hyas SEO v2 twitter:image path problem

### DISCUSSION THREAD

**Discussion by pensivedog** [gethyas/hyas#446] [Q&A]
After upgrading the hyas seo module from v1 to v2, the twitter image path is incorrect. Instead of using, for instance:

`localhost:1313/articles/article-title/article-image.jpg` for the twitter image, it now uses

`localhost:1313/article-image.jpg`, placing the file name directly after the site root, without the rest of the file path.

Any idea how to fix this?

**Comment by pensivedog** [comment]
Doing some further testing on this, it seems the issue is only happening when a page's images are referenced from an `images` array in front matter, which should work regardless of their file names, according to Hugo's [internal template documentation](https://gohugo.io/templates/internal/).

If no such key is defined in front matter AND there is an image with "feature," "cover," or "thumbnail" in its file name, the `twitter:image` path is correct. If an `images` array is defined in front matter, the path is incorrect. 

Is this a bug with Hugo's internal template?

**Reply by h-enk** [reply]
Thanks! Looks like it yes /that could be the case. To be further investigated /for `@hyas/seo` v2.2.0 or to be taken upstream

The [internal template](https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/twitter_cards.html#L1) calls the [get-page-images](https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/partials/_funcs/get-page-images.html) function

**Reply by pensivedog** [reply]
Were you able to reproduce this issue? I also [posted it in the Hugo forum](https://discourse.gohugo.io/t/internal-templates-for-og-image-and-twitter-image-not-working-with-images-array-in-front-matter/47571) but jmooring was unable to replicate the problem.

**Reply by h-enk** [reply]
I can confirm that it indeed works OK when using Joe's repo:

```html
<meta property="og:image" content="http://localhost:1313/posts/post-1/a.jpg" /><meta property="og:image" content="http://localhost:1313/posts/post-1/b.jpg" /><meta property="og:image" content="http://localhost:1313/images/c.jpg" /><meta property="og:image" content="http://localhost:1313/images/d.jpg" />
```

And that it's NOK when using Doks:

```html
<meta property="og:image" content="http://localhost:1313/a.jpg" /><meta property="og:image" content="http://localhost:1313/b.jpg" /><meta property="og:image" content="http://localhost:1313/images/c.jpg" /><meta property="og:image" content="http://localhost:1313/images/d.jpg" />
```

**Reply by h-enk** [reply]
https://github.com/gethyas/seo/issues/16

**Reply by pensivedog** [reply]
Any update on when we can expect a fix for this?

**Reply by h-enk** [reply]
End of this week at the latest

**Reply by h-enk** [reply]
Didn't meet my promise — ouch (other priorities mostly).

But, know that I'm currently working on SEO 2.2 — see also the [Roadmap](https://github.com/orgs/gethyas/discussions/462)

**Reply by pensivedog** [reply]
All good! Appreciate the amazing work you're doing on this.

**Reply by h-enk** [reply]
https://github.com/gethyas/seo/issues/16#issuecomment-1994349590
